v2.0/sql-faqs.md

```diff
@@ -72,6 +90,8 @@ With this setting on, each node of the cluster writes all SQL queries it execute
 > SET CLUSTER SETTING sql.trace.log_statement_execute = false;
 ~~~
 
+### Per-node execution logs
+
 Alternatively, if you are testing CockroachDB locally and want to log queries executed just by a specific node, you can pass `--vmodule=executor=2` to the [`cockroach start`](start-a-node.html) command when starting the node. For example, to start a single node locally and log all SQL queries it executes, you'd run:
```

Comment:
This should also explain how to set `vmodule` using the HTTP endpoint, which does not need to restart the node.

Reply:
Thanks @knz, I did some looking around and discovered this [1.1 beta release notes page](https://www.cockroachlabs.com/docs/releases/v1.1-beta.20170907.html) which says:

> The /debug/vmodule/ endpoint has been removed; use the SQL function crdb_internal.set_vmodule() instead. #17792 #17914

This would lead me to believe that the HTTP endpoint was removed (Also possible that I just don't understand what I am looking at here).  Or is that referring to something else?

I tried doing the following at the SQL prompt as recommended by the release note, but it didn't seem to do anything.  Or at least, it didn't cause the node to start writing my queries to `cockroach.log`:

```
> select crdb_internal.set_vmodule('executor=2');
+---------------------------+
| crdb_internal.set_vmodule |
+---------------------------+
|                         0 |
+---------------------------+
(1 row)
```

I do not understand the return value of zero.

After that, I executed a  statement which I would expect to be logged:

```
> select version();
+--------------------------------------------------------------------------------------+
|                                       version                                        |
+--------------------------------------------------------------------------------------+
| CockroachDB CCL v2.0-beta.20180319 (darwin amd64, built 2018/03/19 18:54:14, go1.10) |
+--------------------------------------------------------------------------------------+
(1 row)
```

Looking in `cockroach.log` for this node right after executing the above, I don't see any SQL logged, just the following:

```
I180402 18:53:01.159587 141 gossip/gossip.go:487  [n1] gossip status (ok, 1 node)
gossip client (0/3 cur/max conns)
gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I180402 18:53:01.161318 143 server/status/runtime.go:219  [n1] runtime stats: 285 MiB RSS, 128 goroutines, 102 MiB/114 MiB/238 MiB GO alloc/idle/total, 88 MiB/127 MiB CGO alloc/total, 72.48cgo/sec, 0.01/0.00 %(u/s)time, 0.00 %gc (0x)
I180402 18:53:11.158430 143 server/status/runtime.go:219  [n1] runtime stats: 286 MiB RSS, 127 goroutines, 108 MiB/108 MiB/238 MiB GO alloc/idle/total, 88 MiB/127 MiB CGO alloc/total, 97.43cgo/sec, 0.01/0.00 %(u/s)time, 0.00 %gc (0x)
I180402 18:53:12.203874 394661 sql/executor.go:2104  [n1,client=127.0.0.1:50155,user=root] query not supported for distSQL: unsupported node *sql.unaryNode
```

Reply:
The vmodule setting for query logging has changed, it is now `exec_log=2` (instead of `executor=2`). This applies both to the command line flag `--vmodule` and this newly discovered sql function.